src/dune_rules/merlin.mli

```diff
@@ -1,11 +1,37 @@
-(** Merlin rules *)
+(** This module ensures that one merlin configuration file is generated for each
+    stanza. Each of these configuration files is accompanied by a merlin exist
+    file. Each of these files contain a map from every module involved in the
+    stanza to a standard merlin configuration. The [Processed.t] type represents
+    the Merlin configuration as it will be marshalled to and from the
+    configuration files, while [Merlin.t] represents raw information coming from
+    the build system. *)
 open! Dune_engine
 
 open! Stdune
 open Import
 
+(** Merlin config folder name *)
+val merlin_folder_name : string
+
+(** Return the name of the merlin file for a given stanza *)
+val make_merlin_exists : Merlin_ident.t -> string
```

Comment:
It would  be nicer if this function was:

```
val merlin_exists_path : dir:Path.t -> Merlin_ident.t -> Path.t
```

You might as well move it to `merlin_ident` as well?